content/en/blog/2024/getting-started-with-otelsql/index.md

```diff
@@ -0,0 +1,426 @@
+---
+title: 'Getting Started with otelsql: OpenTelemetry instrumentation for Go SQL'
+linkTitle: Getting Started with otelsql
+date: 2024-02-24
+author: '[Sam Xie](https://github.com/XSAM) (Cisco)'
+draft: true # TODO: remove this line once your post is ready to be published
+---
+
+[otelsql](https://github.com/XSAM/otelsql) is an instrumentation library for the
+[`database/sql`](https://pkg.go.dev/database/sql) library of Go programming
+language. It generates traces and metrics data while the application interacts
+with the database via `database/sql` library. By that, it allows you to identify
+errors or slowdowns in your SQL queries that potentially impact the performance
+of your application.
+
+This post provides a quick-start guide for using this library.
+
+## Getting Started
+
+otelsql is basically a wrapper layer for interfaces from `database/sql`. When
+users use the wrapped database interfaces, the otelsql generates the telemetry
+data and passes operations to the underlying database.
+
+In the following, you are going to use the docker compose to run an
+otel-collector example from the otelsql repository. This example uses a MySQL
+client with the otelsql instrumentation. The telemetry data it generates will
+push to OpenTelemetry Collector, then it shows the trace data on Jaeger and the
+metrics data on the Prometheus server.
+
+Here is the data flow:
+
+```mermaid
+flowchart LR;
+    A[MySQL client]-->B[OpenTelemetry Collector];
+    B-->C["Jaeger (trace)"];
+    B-->D["Prometheus (metrics)"];
+```
+
+Let's clone the otelsql repository here and run the example. I will walk you
+through some of the most important lines of code.
```

Comment:
```suggestion
Let's clone the otelsql repository here and run the example and take a look at the most important lines of code.
```

This is the only(?) place in the document where you use the "I ..." form, let's change this for consistency.